sql/core/src/main/scala/org/apache/spark/sql/execution/GroupedIterator.scala

```diff
@@ -83,11 +83,12 @@ class GroupedIterator private(
 
   /** Holds a copy of an input row that is in the current group. */
   var currentGroup = currentRow.copy()
-  var currentIterator: Iterator[InternalRow] = null
+
   assert(keyOrdering.compare(currentGroup, currentRow) == 0)
```

Comment:
Looks like we only use `keyOrdering` to do equality check, why not just use `==`? The `currentGroup` and `currentRow` are from the same input, they must be both unsafe or safe, and `==` for `UnsafeRow` is faster than `keyOrdering.compare`.

cc @marmbrus